feat: 친구 DM 전체 레이아웃 및 기능 구현

yeni023 opened the pull request:
## 🔗 반영 브랜치
- closed #133
- `yeeun/friend-dm-ui → dev`

## 📝 작업 내용
- `FriendChatRoom` 전체 레이아웃 구성 및 디자인 반영
- 채팅 입력 시 한글 조합 문제 해결 (중복 메시지 방지)
- 메시지 발신자에 따라 텍스트 색상 구분 (나: `#FFFFFF` / 상대: `#6F59B1`)
- BLOCK 버튼 상태에 따라 텍스트 전환 (BLOCK ↔ UNBLOCK)
- 취소(✕) 버튼 클릭 시 `/Friend` 경로로 이동하도록 구현

### 📸 스크린샷 (선택)
![스크린샷 2025-05-12 오후 5 32 10](https://github.com/user-attachments/assets/bf1ca8fc-276f-4d9d-a2c4-f56b4f174011)

## 💬 리뷰 요구사항(선택)
- 친구 목록에서 상태 표시는 혜주님이 구현하신 부분과 동일해서 받아와서 이어서 작업하겠습니다.
- 친구 목록에서 상대방 프로필 눌렀을 때 해당 채팅이 뜨는 부분도 이어서 작업하겠습니다.

yeni023 committed:
feat: 로그인 페이지 디자인 구현

- vanilla-extract를 사용해 로그인 페이지 스타일 구조 정리
- 기존 px 단위를 rem으로 전환하여 반응형 대응 개선

yeni023 committed:
feat: 친구 DM 디자인 구현

FriendChatRoom 레이아웃 전체 구성 및 스타일링
채팅 한글 입력 시 중복 메시지 방지
내가 보낸 메시지/상대 메시지 색상 구분 (white vs #6F59B1)
BLOCK 버튼 상태에 따른 텍스트 전환 (BLOCK <-> UNBLOCK)
Cancel 버튼 클릭 시 /Friend로 라우팅 이동

HJ0311 commented on src/pages/FriendPage/components/friend-header/styles.css.ts:
```diff
@@ -0,0 +1,46 @@
+export const blockButton = style({
+  background: `url('/assets/images/base-button.png') no-repeat center`,
+  backgroundSize: 'contain',
+  width: '100px',
```
px과 rem을 혼용하고 있어서 rem으로 통일하는게 좋을 것 같아요.

yeni023 replied:
네 확인했습니다 스타일링 작업에서 px로 작업된 부분 rem으로 수정했습니다.

HJ0311 reviewed (dismissed):
수고하셨습니다

hakilee commented on src/pages/FriendPage/components/chat-box/index.tsx:
```diff
@@ -0,0 +1,50 @@
+    <div className={styles.chatBox}>
+      <div className={styles.messages}>
+        {messages.map((msg, idx) => {
+          const isMyMessage = msg.startsWith('ME:');
```
이거 ME로 시작한다는 것이 백엔드에서 정해준 룰인가요?

yeni023 replied:
사용자 ID 가 들어가야 합니다...! 제가 잘못 작성해서 api 연동할 때 사용자 ID 받아오는 걸로 수정해서 작업 진행하겠습니다

hakilee commented on src/pages/FriendPage/components/chat-box/index.tsx:
```diff
@@ -0,0 +1,50 @@
+        {messages.map((msg, idx) => {
+          const isMyMessage = msg.startsWith('ME:');
+          return (
+            <div key={idx} className={isMyMessage ? styles.myMessage : styles.otherMessage}>
```
`idx`로 키를 관리하는 것은 좋은 방법이 아닌 것 같아요. ([참고](https://yozm.wishket.com/magazine/detail/2634/))

yeni023 replied:
올려주신 부분 참고해서 api 연동할 때 이 부분도 같이 수정 진행하겠습니다...!

hakilee commented on src/pages/FriendPage/components/chat-box/index.tsx:
```diff
@@ -0,0 +1,50 @@
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter' && !isComposing) {
```
composing 처리 좋은 것 같습니다 👍

hakilee commented on src/pages/FriendPage/components/chat-box/styles.css.ts:
```diff
@@ -0,0 +1,86 @@
+  flexDirection: 'column',
+  gap: rem(8),
+
+  selectors: {
```
이거 혹시 `global.css.ts` 에 작성된 것과 다른 건가요?

yeni023 replied:
Chat 부분 스크롤바 투명도가 다르길래 따로 했는데 큰차이가 없는 것 같아서 `globla.css.ts`에 정의된 스크롤바 사용하는 걸로 코드 수정하겠습니다.

hakilee commented on src/pages/FriendPage/components/chat-box/styles.css.ts:
```diff
@@ -0,0 +1,86 @@
+  borderRadius: rem(10),
+  padding: `${rem(8)} ${rem(10)}`,
+  margin: `${rem(6)} ${rem(12)} ${rem(12)}`,
+  boxSizing: 'border-box',
```
```suggestion
```

yeni023 replied:
지웠어야 했는데 확인할 때 놓쳤네요... 알려주셔서 감사합니다!

hakilee commented on src/pages/FriendPage/components/friend-chat-room/index.tsx:
```diff
@@ -0,0 +1,35 @@
+  const navigate = useNavigate();
+
+  const handleExit = () => {
+    navigate('/Friend');
```
이거 대문자 맞나요? constants의 `routes.ts` 파일에서 상수를 할당하면 더 좋을 것 같아요.

hakilee commented on src/pages/FriendPage/components/friend-chat-room/styles.css.ts:
```diff
@@ -0,0 +1,62 @@
+  width: rem(200),
+  display: 'flex',
+  flexDirection: 'column',
+  backgroundColor: '#000',
```
```suggestion
  backgroundColor: theme.color.black,
```

hakilee commented on src/pages/FriendPage/components/friend-list/styles.css.ts:
```diff
@@ -0,0 +1,50 @@
+  gap: rem(10),
+  cursor: 'pointer',
+  padding: `${rem(6)} ${rem(1)} ${rem(6)} ${rem(9)}`,
+  borderBottom: `${rem(1.5)} solid #ffffff`,
```
```suggestion
  borderBottom: `${rem(1.5)} solid ${theme.color.white}`,
```

hakilee commented on src/pages/FriendPage/components/friend-list/styles.css.ts:
```diff
@@ -0,0 +1,50 @@
+export const divider = style({
+  width: rem(200),
+  height: rem(1),
+  backgroundColor: '#ffffff',
```
```suggestion
  backgroundColor: theme.color.white,
```

hakilee commented on src/pages/SignInPage/EmailLoginPage/styles.css.ts:
```diff
@@ -1,11 +1,38 @@
+  justifyContent: 'center',
+  alignItems: 'center',
+  marginBottom: rem(12),
+  color: 'white',
```
```suggestion
  color: theme.color.white,
```

yeni023 replied:
`theme.css.ts`에 정의된 색상들 이 형태로 모두 수정하겠습니다!

hakilee commented on src/pages/SignInPage/EmailLoginPage/styles.css.ts:
```diff
@@ -1,11 +1,38 @@
+  background: 'none',
+  border: 'none',
+  borderBottom: 'none',
+  color: 'white',
```
```suggestion
  color: theme.color.white,
```

hakilee commented on src/pages/SignInPage/EmailLoginPage/styles.css.ts:
```diff
@@ -1,11 +1,38 @@
+export const input = style({
+  background: 'none',
+  border: 'none',
+  borderBottom: 'none',
```
```suggestion
```

`border`과 중복

hakilee commented on src/pages/index.ts:
```diff
@@ -1,4 +1,5 @@
 export * from './FriendPage';
+export * from './FriendPage/FriendChatRoom';
```
이게 routes 에는 dm으로 되어있는 것 같은데, 용어를 하나로 만드는게 좋을 것 같습니다.

yeni023 replied:
하나의 용어로 통일하는 게 좋을 것 같아서 `/friend/chatroom` 으로 변경했습니다.

hakilee commented on src/pages/FriendPage/FriendChatRoom.tsx:
```diff
@@ -0,0 +1,5 @@
+import { FriendChatRoom } from './components/friend-chat-room';
+
+export const FriendChatRoomPage = () => {
+  return <FriendChatRoom />;
```
`FriendChatRoom` 컴포넌트를 호출하는 것보다, 여기에 FriendChatRoom 내부 컴포넌트를 나열하는게 더 가독성이 좋지 않을까 싶습니다. (deps가 하나 더 생기니까요.)

yeni023 replied:
확인했습니다! 말씀해주신대로 작업하는 게 가독성이 더 좋아보여서 수정해서 PR 올리겠습니다

hakilee reviewed (commented):
고생하셨습니다. 몇 가지 코멘트 남겼는데 확인해 주세요.

추가로, `#333`, `#FD906F`과 같은 값들이 여러번 사용되는데, `theme.css.ts` 파일의 color 값에서 관리하는 방식으로 가면 더 좋을 것 같아요.

yeni023 committed:
chore: PR 리뷰 기반 스타일링 및 구조 개선

- theme 변수로 색상 상수화 (e.g. #FD906F → theme.color.orangeChat)
- 불필요한 rem → px 혼용 제거 및 일관성 유지
- 불필요하거나 중복되는 스타일링 제거
- 라우트 명명 규칙 통일 (/friend/dm → /friend/chatroom)
- 컴포넌트 분리로 의존성 감소 및 가독성 향상

yeni023 committed:
Merge branch 'dev' into yeeun/friend-dm-ui

hakilee reviewed (approved):
고생하셨습니다! 💯
